build-support/scripts/protobuf.sh

```diff
@@ -45,54 +44,29 @@ function main {
                 usage
                 return 0
                 ;;
-            --protoc-version )
-                protoc_version="$2"
-                shift 2
-                ;;
             --tools-only )
                 tools_only=1
                 shift
                 ;;
         esac
     done
 
-    if test -z "${protoc_version}"
-    then
-        protoc_version="$(make --no-print-directory print-PROTOC_VERSION)"
-        if test -z "${protoc_version}"
-        then
-            err_usage "ERROR: no proto-version specified and version could not be discovered"
-            return 1
-        fi
-    fi
-
-    # ensure the correct protoc compiler is installed
-    protoc_install "${protoc_version}"
-    if test -z "${protoc_bin}" ; then
-        exit 1
-    fi
-
     # ensure these tools are installed
     proto_tools_install
 
     if [[ -n $tools_only ]]; then
         return 0
     fi
 
-    # Compute some data from dependencies in non-local variables.
-    go mod download
-    golang_proto_path="$(go list -f '{{ .Dir }}' -m github.com/golang/protobuf)"
-    # golang_proto_mod_path="$(sed -e 's,\(.*\)github.com.*,\1,' <<< "${golang_proto_path}")"
-    golang_proto_mod_path="$(go env GOMODCACHE)"
-
-    declare -a proto_files
-    while IFS= read -r pkg; do
-        pkg="${pkg#"./"}"
-        proto_files+=( "$pkg" )
-    done < <(find . -name '*.proto' | grep -v 'vendor/' | grep -v '.protobuf' | sort )
-
-    for proto_file in "${proto_files[@]}"; do
-        generate_protobuf_code "${proto_file}"
+    for mod in $(find . -name 'buf.gen.yaml' -exec dirname {} \; | sort)
+    do
+        pushd $mod
+        buf generate
+        for proto_file in $(buf ls-files)
+        do
+            postprocess_protobuf_code $proto_file
+        done
+        popd
```

Comment:
```suggestion
        (
            cd $mod
            buf generate
            for proto_file in $(buf ls-files)
            do
                postprocess_protobuf_code $proto_file
            done
        )
```
I usually prefer to use a subshell rather than pushd/popd these days since it's self-correcting (like a `defer` would be). I've been burned in the past by a missed `popd` that let a loop wildly astray.